modules/margin-protocol/src/lib.rs

```diff
@@ -788,6 +825,46 @@ impl<T: Trait> Module<T> {
 			Ok(())
 		}
 	}
+
+	/// Force closure position to liquidate liquidity pool based on opened positions.
+	///
+	/// Return `Ok` if closure success, or `Err` if not.
+	fn _pool_liquidate(pool: LiquidityPoolId, position_id: PositionId) -> DispatchResult {
+		let position = Self::positions(position_id).ok_or(Error::<T>::PositionNotFound)?;
+
+		let price = Self::_price(position.pair.base, position.pair.quote)?;
+		let spread = {
+			if position.leverage.is_long() {
+				T::LiquidityPools::get_bid_spread(pool, position.pair)
+					.ok_or(Error::<T>::NoAskSpread)?
+					.into()
+			} else {
+				T::LiquidityPools::get_ask_spread(pool, position.pair)
+					.ok_or(Error::<T>::NoAskSpread)?
+					.into()
+			}
+		};
+
+		// Profit from spread + penalty
+		let price_spread = fixed_128_from_fixed_u128(price)
+			.saturating_mul(spread)
+			.saturating_mul(Fixed128::from_parts(2));
+
+		let spread_profit = position
+			.leveraged_held
+			.checked_mul(&price_spread)
+			.ok_or(Error::<T>::NumOutOfBound)?;
+
+		let profit_in_usd = Self::_usd_value(position.pair.base, spread_profit)?;
```

Comment:
Pls refactor these lines, the naming here is confusing: your `price_spread` is not actual spread, `spread_profit` is not profit, and `profit_in_usd` is not profit in USD either.